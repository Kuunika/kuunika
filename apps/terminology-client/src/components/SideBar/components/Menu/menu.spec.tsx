import React from 'react';
import { render } from '@testing-library/react';

import { MenuView } from './index';
import Store from '../../../../services/redux/store';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';

describe('Menu', () => {
  it('should render all menu items', () => {
    const data = {
      items: [
        {
          name: 'clinical',
          displayName: 'Clinical',
          link: './clinical',
          icon: ['fas fa-ambulance']
        },
        {
          name: 'Menu2',
          displayName: 'Menu2',
          link: './menu2',
          icon: ['fas fa-user-md']
        },
        {
          name: 'Menu3',
          displayName: 'Menu3',
          link: './menu3',
          icon: ['fas fa-pills']
        }
      ]
    };
    const { baseElement } = render(
      <Provider store={Store}>
        <BrowserRouter>
          <MenuView {...data} loading={false} />
        </BrowserRouter>
      </Provider>
    );
    expect(baseElement).toMatchSnapshot();
  });
});
