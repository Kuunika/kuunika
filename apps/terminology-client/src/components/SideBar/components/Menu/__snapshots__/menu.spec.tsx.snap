// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should render all menu items 1`] = `
<body>
  <div>
    <div
      class="sc-htpNat jmINdp"
      data-testid="menu-container"
    >
      <a
        data-testid="menu-clinical"
        href="/clinical"
        style="text-decoration: none;"
      >
        <div
          class="sc-bdVaJa lmvtnY"
        >
          Clinical
        </div>
      </a>
      <a
        data-testid="menu-Menu2"
        href="/menu2"
        style="text-decoration: none;"
      >
        <div
          class="sc-bdVaJa jWaYE"
        >
          Menu2
        </div>
      </a>
      <a
        data-testid="menu-Menu3"
        href="/menu3"
        style="text-decoration: none;"
      >
        <div
          class="sc-bdVaJa jWaYE"
        >
          Menu3
        </div>
      </a>
    </div>
  </div>
</body>
`;
